hybrid/pkg/agent/agent_interceptor.go

```diff
@@ -0,0 +1,93 @@
+package hybrid_agent
+
+import (
+	"context"
+	"encoding/json"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hewlettpackard/hybrid/pkg/common"
+	nodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type AgentInterceptorInterface interface {
+	Recv() (*nodeattestorv1.Challenge, error)
+	Send(challenge *nodeattestorv1.PayloadOrChallengeResponse) error
+	setCustomStream(stream nodeattestorv1.NodeAttestor_AidAttestationServer)
+	SetContext(ctx context.Context)
+	Context() context.Context
+	SetLogger(logger hclog.Logger)
+	SendCombined(common.PluginMessageList) error
+	SetPluginName(name string)
+	GetMessage() common.PluginMessage
+	SpawnInterceptor() AgentInterceptorInterface
+}
+
+type HybridPluginAgentInterceptor struct {
+	ctx    context.Context
```

Comment:
I cannot remove the context because the interceptor must conform with the grpc.ServerStream interface, so we don't have any problems with his injection

Reply:
but, you have stream as an attribute of `HybridPluginAgentInterceptor`  can not you get ctx from there?